tgui/packages/tgui/interfaces/LogViewer.tsx

```diff
@@ -0,0 +1,231 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Collapsible, Input, NoticeBox, Section, Stack } from '../components';
+import { Window } from '../layouts';
+
+type LogViewerData = {
+  round_id: number;
+  logging_start_timestamp: string;
+  tree: LogViewerCategoryTree;
+  last_data_update: number;
+  categories: Record<string, LogViewerCategoryData>;
+};
+
+type LogViewerCategoryTree = {
+  enabled: string[];
+  disabled: string[];
+};
+
+type LogViewerCategoryData = {
+  entry_count: number;
+  entries: LogEntryData[];
+};
+
+type LogEntryData = {
+  id: number;
+  message: string;
+  timestamp: string;
+  semver?: Record<string, string>;
+  data?: any[];
+};
+
+export const LogViewer = (_: any, context: any) => {
+  const { data, act } = useBackend<LogViewerData>(context);
+
+  const [activeCategory, setActiveCategory] = useLocalState(
+    context,
+    'activeCategory',
+    ''
+  );
+
+  return (
+    <Window width={720} height={720}>
+      <Window.Content scrollable>
+        <Section>
+          <Button icon="sync" onClick={() => act('refresh')} />
+        </Section>
+        <CategoryBar
+          options={data.tree.enabled}
+          active={activeCategory}
+          setActive={setActiveCategory}
+        />
+        <CategoryViewer
+          activeCategory={activeCategory}
+          data={data.categories[activeCategory]}
+        />
+      </Window.Content>
+    </Window>
+  );
+};
+
+type CategoryBarProps = {
+  options: string[];
+  active: string;
+  setActive: (active: string) => void;
+};
+
+const CategoryBar = (props: CategoryBarProps, context: any) => {
+  const sorted = [...props.options].sort();
+  const [categorySearch, setCategorySearch] = useLocalState(
+    context,
+    'categorySearch',
+    ''
+  );
+
+  return (
+    <Section
+      title="Categories"
+      scrollableHorizontal
+      buttons={
+        <Input
+          grow
+          placeholder="Search"
+          value={categorySearch}
+          onChange={(_: any, value: string) => setCategorySearch(value)}
+        />
+      }>
+      <Stack scrollableHorizontal>
+        {sorted.map((category) => {
+          if (!category.toLowerCase().includes(categorySearch.toLowerCase())) {
+            return null;
+          }
+          return (
+            <Stack.Item key={category}>
+              <Button
+                textAlign="left"
+                content={category}
+                selected={category === props.active}
+                onClick={() => props.setActive(category)}
+              />
+            </Stack.Item>
+          );
+        })}
+      </Stack>
+    </Section>
+  );
+};
+
+type CategoryViewerProps = {
+  activeCategory: string;
+  data?: LogViewerCategoryData;
+};
+
+const validateRegExp = (str: string) => {
+  try {
+    new RegExp(str);
+    return true;
+  } catch (e) {
+    return false;
+  }
```

Comment:
Why not print the exception message somewhere? It could help identify a typo in the regex easily, without needing external tools. IE doesn't support some regex features  either last I checked, so this would be helpful